nimbus/src/nimbus.idl

```diff
@@ -78,6 +80,16 @@ interface NimbusClient {
         AvailableRandomizationUnits available_randomization_units
     );
 
+    // Initializes the database and caches enough information so that the
+    // non-blocking API functions (eg, `get_experiment_branch()`) can
+    // return accurate results rather than throwing a "not initialized" error.
+    // It's not strictly necessary to call this function - any function that
+    // wants to use the database will implicitly initialize too - this exists
+    // so that the consuming application can have confidence the non-blocking
+    // functions will return data instead of returning an error, and will do
+    // the minimum amount of work to achieve that.
+    [Throws=Error]
+    void initialize();
```

Comment:
Nit: `initialize()` sounds like it is necessary (your docs make it's clear that it's not).

When should I use this, rather than `applyPendingExperiments()`? Testing is a reasonable answer, btw.

Reply:
As the comment says, whenever you want to do the *least amount of IO* which allows `getExperimentBranch()` to give you an answer. `applyPendingExperiments()` does write IO, so it seems quite clear that Fenix would *not* want to call that if it intends using `getExperimentBranch()` early in startup.

I believe suggesting `applyPendingExperiments()` at startup is going to be an anti-pattern for all the "avoid early IO" discussions we've been having.

Reply:
I've added your new `initialize()` method to https://github.com/mozilla-mobile/android-components/pull/9394 .

> I believe suggesting applyPendingExperiments() at startup is going to be an anti-pattern for all the "avoid early IO" reasons we've been discussing.

I think at that point, we'd want to optimize the `apply_pending_experiments` + `update_cache` combo to calculate the cache before flushing it to disk. We've had a lot of push back from Fenix to avoid enrollment changes within the same session.